selfservice/strategy/password/login.go

```diff
@@ -81,17 +83,33 @@ func (s *Strategy) Login(w http.ResponseWriter, r *http.Request, f *login.Flow,
 		return nil, herodot.ErrInternalServerError.WithReason("The password credentials could not be decoded properly").WithDebug(err.Error()).WithWrap(err)
 	}
 
-	if err := hash.Compare(r.Context(), []byte(p.Password), []byte(o.HashedPassword)); err != nil {
-		return nil, s.handleLoginError(w, r, f, &p, errors.WithStack(schema.NewInvalidCredentialsError()))
-	}
+	if o.ShouldUsePasswordMigrationHook() {
```

Comment:
Who sets this to false once this is executed successfully? Is it `s.migratePasswordHash`?

Reply:
Looks like it is

Reply:
Yes, if the password hash is not empty, this returns false.